framework/include/fvbcs/FVFluxBC.h

```diff
@@ -0,0 +1,35 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "FVBoundaryCondition.h"
+
+/**
+ * Base class for Finite Volume Flux BCs
+ */
+class FVFluxBC : public FVBoundaryCondition
+{
+public:
+  FVFluxBC(const InputParameters & parameters);
+
+  static InputParameters validParams();
+
+  virtual void computeResidual(const FaceInfo & fi);
+  virtual void computeJacobian(const FaceInfo & fi);
+
+protected:
+  virtual ADReal computeQpResidual() = 0;
+
+  const unsigned int _qp = 0;
```

Comment:
We don't really need it.  I kept it in case we ever decide to do higher order integration with higher order reconstructed solutions, and also because it makes FV kernels/bcs look and feel more like FE ones.  If we don't think those are good enough reasons we can remove it.  There is also the issue of coupling between FE and FV - do we expect FE objects to know when they should be qp indexing vs not depending on the type of variable they couple to?  Also do we want to support input-file-toggling between FE and FV vars for some parts of a problem? - if yes, then we need to keep the qp indexing.